pkg/controller/v1beta1/inferenceservice/reconcilers/keda/keda_reconciler.go

```diff
@@ -0,0 +1,212 @@
+)
+
+var log = logf.Log.WithName("KedaReconciler")
+
+type KedaReconciler struct {
+	client       client.Client
+	scheme       *runtime.Scheme
+	ScaledObject *kedav1alpha1.ScaledObject
+	componentExt *v1beta1.ComponentExtensionSpec
+}
+
+func NewKedaReconciler(client client.Client,
+	clientset kubernetes.Interface,
+	scheme *runtime.Scheme,
+	componentMeta metav1.ObjectMeta,
+	componentExt *v1beta1.ComponentExtensionSpec) *KedaReconciler {
+	return &KedaReconciler{
+		client:       client,
+		scheme:       scheme,
+		ScaledObject: createKedaScaledObject(clientset, componentMeta, componentExt),
+		componentExt: componentExt,
+	}
+}
+
+func getKedaMetrics(metricConfig *v1beta1.MetricsConfig, metadata metav1.ObjectMeta,
+	componentExt *v1beta1.ComponentExtensionSpec) []kedav1alpha1.ScaleTriggers {
+	var triggers []kedav1alpha1.ScaleTriggers
+	var serverAddress string
+
+	// Default values
+	triggerType := string(corev1.ResourceCPU)
+	metricType := autoscalingv2.UtilizationMetricType
+	scaleTarget := int(constants.DefaultCPUUtilization)
+
+	// Get metric configuration from metricConfig (configmap)
+	if metricConfig != nil && metricConfig.MetricBackend != "" {
+		triggerType = metricConfig.MetricBackend
+		serverAddress = metricConfig.ServerAddress
+	}
+	// override metric configuration from componentExtension if it is set
+	if componentExt.ScaleMetric != nil {
+		triggerType = string(*componentExt.ScaleMetric)
+	}
+	if componentExt.ScaleMetricType != nil {
+		metricType = *componentExt.ScaleMetricType
+	}
+	if componentExt.ScaleTarget != nil {
+		scaleTarget = *componentExt.ScaleTarget
+	}
+	// override metric configuration from componentExtension.ScalerSpec if it is set
+	if componentExt.ScalerSpec != nil {
+		if componentExt.ScalerSpec.ScaleMetric != nil {
+			triggerType = string(*componentExt.ScalerSpec.ScaleMetric)
+		}
+		if componentExt.ScalerSpec.ServerAddress != "" {
+			serverAddress = componentExt.ScalerSpec.ServerAddress
+		}
+		if componentExt.ScalerSpec.ScaleMetricType != nil {
+			scaleTarget = *componentExt.ScalerSpec.ScaleTarget
+		}
+		if componentExt.ScalerSpec.ScaleMetricType != nil {
+			metricType = *componentExt.ScalerSpec.ScaleMetricType
+		}
+	}
+
+	trigger := kedav1alpha1.ScaleTriggers{
+		Type:     triggerType,
+		Metadata: map[string]string{},
+	}
+
+	// set trigger metadata for prometheus and graphite triggers
+	if triggerType == "prometheus" || triggerType == "graphite" {
+		if serverAddress != "" {
+			trigger.Metadata["serverAddress"] = serverAddress
+		}
+		if componentExt.ScalerSpec != nil {
+			if componentExt.ScalerSpec.MetricQuery != "" {
+				trigger.Metadata["query"] = componentExt.ScalerSpec.MetricQuery
+			}
+			if componentExt.ScalerSpec.QueryParameters != "" {
+				trigger.Metadata["queryParameters"] = componentExt.ScalerSpec.QueryParameters
+			}
+			if componentExt.ScalerSpec.ServerAddress != "" {
+				trigger.Metadata["threshold"] = strconv.Itoa((scaleTarget))
```

Comment:
Any plan to support authentication refs? Downstream we use Prometheus and Thanos so to make this work we need that by default.